python/ray/dashboard/subprocesses/utils.py

```diff
@@ -12,17 +12,38 @@ class ResponseType(enum.Enum):
     WEBSOCKET = "websocket"
 
 
-def module_logging_filename(module_name: str, logging_filename: str) -> str:
+def module_logging_filename(
+    module_name: str, incarnation: int, logging_filename: str
+) -> str:
     """
     Parse logging_filename = STEM EXTENSION,
-    return STEM _ MODULE_NAME EXTENSION
+    return STEM _ MODULE_NAME _ INCARNATION EXTENSION
+
+    If logging_filename is empty, return "stderr"
 
     Example:
     module_name = "TestModule"
+    incarnation = 5
     logging_filename = "dashboard.log"
     STEM = "dashboard"
     EXTENSION = ".log"
-    return "dashboard-TestModule.log"
+    return "dashboard_TestModule_5.log"
     """
+    if not logging_filename:
+        return "stderr"
     stem, extension = os.path.splitext(logging_filename)
-    return f"{stem}_{module_name}{extension}"
+    return f"{stem}_{module_name}_{incarnation}{extension}"
+
+
+def get_socket_path(socket_dir: str, module_name: str) -> str:
+    socket_path = os.path.join(socket_dir, "dashboard_" + module_name)
+    # The max length of a Unix socket path is 108 bytes.
+    if len(socket_path) > 108:
+        # Discard the "dashboard_" prefix.
+        socket_path = os.path.join(socket_dir, module_name)
+    if len(socket_path) > 108:
+        raise ValueError(
+            f"Socket path {socket_path} is too long. "
+            "Please use a shorter module name or a shorter socket_dir."
+        )
+    return socket_path
```

Comment:
In tests, the `socket_dir` is very long because it is the `tmp_path` pytest fixture plus the session folder name, which contains a timestamp. It looks like this:

`/tmp/4163fbc40ed74552a12e343de9b35fc9/session_2025-03-14_11-09-51_033910_1341615/sockets/`

The maximum length for a Unix socket is 108 bytes. Therefore, if the total `socket_path` exceeds 108 characters, we discard the `dashboard_` prefix.